### PR TITLE
Refacto code

### PR DESCRIPTION
**Pull request by 3rdCore**
This is my proposition for refactoring SeaPearl.jl include order. 
It doesn't improve SeaPearl.jl compilation performance as described in https://docs.julialang.org/en/v1/manual/code-loading/, but makes the code cleaner and more understandable.

I also fix a little bug on the metrics plotting function :  `plotRewardVariation`

### CODE REVIEW

File: src/SeaPearl.jl [hunk 1]
```diff
@@ -13,9 +14,4 @@ include("experiment/experiment.jl")
 
 greet() = print("Hello World!")
 
-export
-
-#Graph
-edges, has_edge, nv, ne, CPLayerGraph
```
Comment (3rdCore) [inline src/SeaPearl.jl:19]: I don't understand the utility of this line.

File: src/experiment/metrics/basicmetrics.jl [hunk 2]
```diff
@@ -235,7 +238,7 @@ end
 plot the relative scores ( compared to the optimal ) of the heuristic during the search for fixed instances along the training. This plot is 
 meaningful only if the metrics is one from the evaluator (ie. the instance remains the same one).
 """
-function plotRewardVariation(metrics::BasicMetrics{TakeObjective, LearnedHeuristic{SR,R,A}}; filename::String="") where {SR, R, A}
+function plotRewardVariation(metrics::BasicMetrics{<:AbstractMetrics, LearnedHeuristic{SR,R,A}}; filename::String="") where {SR, R, A}
```
Comment (3rdCore) [inline src/experiment/metrics/basicmetrics.jl:241]: We should be able to plot the reward evolution of a learned heuristic relatively to any kind of problem ( considering an Objective or not considering an Objective ).

**Review by tomsander1998 — APPROVED** [review]
clean!

**Review by tomsander1998 — APPROVED** [review]
seems good to me

**Pull request merged**